pkgs/misc/emulators/retroarch/default.nix

```diff
@@ -43,11 +45,13 @@ in stdenv.mkDerivation rec {
                 ++ optional enableNvidiaCgToolkit nvidia_cg_toolkit
                 ++ optional withVulkan [ vulkan-loader ]
                 ++ optionals stdenv.isDarwin [ libobjc AppKit Foundation ]
-                ++ optionals stdenv.isLinux [ alsaLib libpulseaudio libv4l libX11
-                                              libXdmcp libXext libXxf86vm udev ];
+                ++ optionals stdenv.isLinux [ alsaLib libdrm libpulseaudio libv4l libX11
+                                              libXdmcp libXext libXxf86vm mesa udev ];
 
   enableParallelBuilding = true;
 
+  configureFlags = if stdenv.isLinux then [ "--enable-kms" ] else "";
```

Comment:
@kolbycrouch for future reference please use `lib.optional` or `lib.optional`, as in:
```
configureFlags = lib.optional stdenv.isLinux "'-enable-kms";
```
or
```
configureFlags = lib.optionals stdenv.isLinux [ "--enable-kms" ];
```